main.go

```diff
@@ -1,62 +1,284 @@
+		},
+		&sensu.PluginConfigOption{
+			Path:     "cacert",
+			Env:      "PUPPET_CACERT",
+			Argument: "cacert",
+			Usage:    "path to the site's Puppet CA certificate PEM file",
+			Value:    &handler.puppetCACert,
+		},
+		&sensu.PluginConfigOption{
+			Path:     "insecure-skip-tls-verify",
+			Env:      "PUPPET_INSECURE_SKIP_TLS_VERIFY",
+			Argument: "insecure-skip-tls-verify",
+			Usage:    "skip SSL verification",
+			Value:    &handler.puppetInsecureSkipVerify,
+		},
+		{
+			Path:      "sensu-api-url",
+			Env:       "SENSU_API_URL",
+			Argument:  "sensu-api-url",
+			Shorthand: "u",
+			Default:   "http://localhost:8080",
+			Usage:     "The Sensu API URL",
+			Value:     &handler.sensuAPIURL,
+		},
+		{
+			Path:      "sensu-api-key",
+			Env:       "SENSU_API_KEY",
+			Argument:  "sensu-api-key",
+			Shorthand: "a",
+			Usage:     "The Sensu API key",
+			Value:     &handler.sensuAPIKey,
+		},
+		{
+			Path:      "sensu-ca-cert",
+			Env:       "SENSU_CA_CERT",
+			Argument:  "sensu-ca-cert",
+			Shorthand: "c",
+			Usage:     "The Sensu Go CA Certificate",
+			Value:     &handler.sensuCACert,
 		},
-	}
-
-	options = []*sensu.PluginConfigOption{
-		&handlerConfigOptions.Example,
 	}
 )
 
 func main() {
-	handler := sensu.NewGoHandler(&handlerConfig.PluginConfig, options, checkArgs, executeHandler)
+	handler := sensu.NewGoHandler(&handler.PluginConfig, options, validate, executeHandler)
 	handler.Execute()
 }
 
-func checkArgs(_ *types.Event) error {
-	if len(handlerConfig.Example) == 0 {
-		return fmt.Errorf("--example or HANDLER_EXAMPLE environment variable is required")
+func validate(event *types.Event) error {
+	// Make sure we have a valid event
+	if event.Check == nil || event.Entity == nil {
+		return errors.New("invalid event")
+	}
+
+	// Make sure all required options are provided
+	if len(handler.endpoint) == 0 {
+		return errors.New("the PuppetDB API endpoint is required")
 	}
+	if len(handler.puppetCert) == 0 {
+		return errors.New("the path to the SSL certificate is required")
+	}
+	if len(handler.puppetKey) == 0 {
+		return errors.New("the path to the private key is required")
+	}
+	if len(handler.sensuAPIURL) == 0 {
+		return errors.New("the Sensu API URL is required")
+	}
+	if len(handler.sensuAPIKey) == 0 {
+		return errors.New("the Sensu API key is required")
+	}
+
+	// Make sure the PuppetDB endpoint URL is valid
+	u, err := url.Parse(handler.endpoint)
+	if err != nil {
+		return fmt.Errorf("invalid PuppetDB API endpoint URL: %s", err)
+	}
+	if u.Scheme == "" || u.Host == "" {
+		return errors.New("invalid PuppetDB API endpoint URL")
+	}
+	if u.Path == "" || u.Path == "/" {
+		u.Path = path.Join(u.Path, defaultAPIPath)
+		handler.endpoint = u.String()
+	}
+
+	// Make sure the Sensu API URL is valid
+	u, err = url.Parse(handler.sensuAPIURL)
+	if err != nil {
+		return fmt.Errorf("invalid Sensu API URL: %s", err)
+	}
+	if u.Scheme == "" || u.Host == "" {
+		return errors.New("invalid Sensu API URL")
```

Comment:
In this case you definitely need to know the scheme; but maybe you can separate the 2 conditions into 2 branches, so that you can give a crystal clear error message if the scheme is missing?